### DISCUSSION TITLE
How to access Windows.Services.Store;

### DISCUSSION THREAD

**Discussion by mpulukkinen** [AvaloniaUI/Avalonia#17640] [Q&A]
### Is your feature request related to a problem? Please describe.

I've literally exhausted all my knowledge trying to find anser to this? How to get access to this api: using Windows.Services.Store;
In maui, I'd target net8.0-windows10.0.19041.0 and that gives me what I need. In avalonia (net9) it just completely breaks the project with hunders of errors. 

I almost thought I succeeded by creating new librry and mimicing this: https://github.com/EmilAlipiev/XFInAppBilling But the app crashes when I try to start it: The program '[27668] LyricVideoStudio.exe' has exited with code 2147516553 (0x80008089).

As a background info, I got fed up with MAUI and I'm converting my app to Avalonia :)

### Describe the solution you'd like

Multiplatform way of accessing stores. What I especially need from that API is GetAppLicenseAsync, GetAppAndOptionalStorePackageUpdatesAsync (because by default, microsoft store does not update the apps automatically), RequestRateAndReviewAppAsync and RequestPurchaseAsync

**Comment by maxkatz6** [comment]
We won't provide a cross-platform APIs for store out of the box. It can be done as a third-party library, if anybody wants to do that.

> net8.0-windows10.0.19041.0

This approach can work in Avalonia too. Just don't forget to either create a dedicated windows project or use multitarget.

**Comment by mpulukkinen** [comment]
Ah ok, thanks for the clarification. I prototyped a bit with the sample cross platform project, so it looks like what I need is one project with output type of WinExe and TargetFramework with net9.0-windows10.0.19041.0

Then the rest of the ui must be in project targetin only net9. 

What I had was just one project with all in it (not counting my data objects class library) and that's why it immediately made a huge mess when I tried to target it with net9.0-windows10.0.19041.0.

So, lesson learned and tomorrow I'm gonna split my project to even smaller pieces 👍
